sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -0,0 +1,112 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.spark.sql.execution.command
+
+import java.net.URI
+
+import scala.util.control.NonFatal
+
+import org.apache.hadoop.fs.{FileSystem, Path}
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.catalog.{CatalogStatistics, CatalogTable}
+import org.apache.spark.sql.internal.SessionState
+
+
+object CommandUtils extends Logging {
+
+  /**
+   * Update statistics (currently only sizeInBytes) after changing data by commands.
+   */
+  def updateTableStats(
+      sparkSession: SparkSession,
+      table: CatalogTable,
+      newTableSize: Option[BigInt] = None,
+      newRowCount: Option[BigInt] = None): Unit = {
+    if (sparkSession.sessionState.conf.autoStatsUpdate && table.stats.nonEmpty) {
+      val catalog = sparkSession.sessionState.catalog
+      val newTable = catalog.getTableMetadata(table.identifier)
+      val newSize = newTableSize.getOrElse(
+        CommandUtils.calculateTotalSize(sparkSession.sessionState, newTable))
+      catalog.alterTableStats(table.identifier,
+        CatalogStatistics(sizeInBytes = newSize, rowCount = newRowCount))
```

Comment:
In my initial design, I planned to update size by default and update other stats based on a flag.
Auto update all stats seems too aggressive to me with one flag...